utilities/generate_api.py

```diff
@@ -36,18 +36,33 @@
 }
 
 
-def run_generate_api(config_path, noisy=False):
+def run_generate_api(config_path, artifact_type, root_dir=None, has_proto_libs=True, noisy=False):
+    """ Generate an API client library.
+
+    :param config_path: (str) Path to directory containing artman config file.
+    :param artifact_type: (str) artman target, e.g "java_gapic".
+    :param root_dir: (str) Path to the directory containing the API definitions.
+    :param has_proto_libs: (bool) if the generated proto and grpc libraries
+              should replace the existing ones
+    :param noisy: (bool) if console output should be verbose.
+
+    """
     googleapis_index = config_path.rfind('/google/')
     if googleapis_index == -1:
         raise ValueError('Didn\'t find /googleapis/ in config file path; need absolute path to the artman config file.')
-    root_dir = config_path[0:googleapis_index]
-    api_dir = config_path[googleapis_index+1:]
+    if has_proto_libs:
+        root_dir = config_path[0:googleapis_index]
+        api_dir = config_path[googleapis_index+1:]
+    else:
+        api_dir = config_path
 
     extra_options = []
     if noisy:
         extra_options = ['-v']
 
-    subprocess.check_call(['artman', '--config', api_dir, '--local', '--root-dir', root_dir] + extra_options + ['generate', 'java_gapic'])
+    print("api_dir: "+ api_dir)
+    print("root_dir: "+ root_dir)
```

Comment:
Are these print outs just for you to debug? If you want to keep them, only print them if `noisy` is true.

Reply:
Removed.